server/routerlicious/packages/lambdas/src/utils/connectionCountLogger.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ICache } from "@fluidframework/server-services-core";
+import { LumberEventName, Lumberjack } from "@fluidframework/server-services-telemetry";
+
+export interface IConnectionCountLogger {
+    /**
+     * This function will store total connection count per node and per cluster in Redis, and
+     * will also log these counters using ConnectionCountPerNode and TotalConnectionCount metrics respectively.
+     */
+    logConnectionCount(isConnect: boolean): any;
+}
+
+export class ConnectionCountLogger implements IConnectionCountLogger {
+    constructor(private readonly nodeName: string = "nodeName", private readonly cache: ICache | undefined) {
+    }
+
+    public async logConnectionCount(isConnect: boolean): Promise<void> {
```

Comment:
Instead of using a boolean to indicate if we are adding a connection or removing a connection from the count, would it be possible to instead provide two public methods here, one called `increaseConnectionCount()` and other called `decreaseConnectionCount()`?

Reply:
I had that before but the implementation was very similar. so decided to keep it in one.

Reply:
I think it might be easier to understand what connectionCountLogger is doing in Alfred's index.ts if we use more specific names. The implementation is similar, but in the current implementation, there is not much code that gets reused anyway. One idea would be to have `increaseConnectionCount()` and `decreaseConnectionCount()` internally call a private function with the shared logic. That way, you don't mix internal implementation with the external view

Reply:
on it.